gas/testsuite/gas/riscv/zfa.s

```diff
@@ -0,0 +1,92 @@
+target:
+	# fli: test both decimal and symbol representations
+	#      (0..29, min==1, inf==(30), nan==(31))
+	fli.h		ft1, 0
```

Comment:
hi，I have some doubts about the ```fli``` instruction.
I'm not sure. If the number 0-31 is used instead of the table "Immediate values loaded by the FLI.S instruction", how can this reflect the actual floating point of the load at the instruction level? Is it more appropriate to use ```fli.h ft1, -1.0``` instead of ```fli.h ft1, 0```?

Reply:
Hi. This extension implements recommended assembly syntax described in Chapter 25 of [the latest RISC-V ISA Manual draft](https://github.com/riscv/riscv-isa-manual/releases/tag/draft-20221217-cb3b9d1) with a GNU extension (not just plain decimal constants, it accepts all kinds of expression as long as the result is an integer constant (without relocation) within range of 0..29.

In the recommended syntax, following entries should be acceptable:
* Decimal constant 0 through 29 (in C-like syntax)
* `min` (for entry 1)
* `inf` (for entry 30)
* `nan` (for entry 31)

And actual immediate loaded by this instruction is described in "25.1 Load-Immediate Instructions".

I think floating point constant operand will make the syntax outside of the recommendation and complicate the assembler in various ways.

Reply:
I forgot to note that, I confirmed this syntax directly with Dr. Andrew Waterman, the main author of this extension draft (when a part of the `Zfa` extension containing `FLI` instructions is called `Zfb`) and I'm sure that this (current syntax implemented) is OK (meaning that a numeric constant or a special operand corresponds to a table entry).

Reply:
Yes, I can understand. This way of implementation is feasible, but I think it may be more appropriate to embody the specific decimal floating-point number at the assembly level. 

This has a similar example on arm architecture.
[ARMv8 Instruction Set Overview](https://www.cs.princeton.edu/courses/archive/spr19/cos217/reading/ArmInstructionSetOverview.pdf)
> 5.6.3 Floating-point Move (immediate) 
The floating point constant fpimm may be specified either in decimal notation (e.g. “12.0” or “-1.2e1”), or as a 
string beginning “0x” followed by the hexadecimal representation of its IEEE754 encoding. A disassembler should 
prefer the decimal notation, so long as the value can be displayed precisely.  
The floating point value must be expressable as ±n÷16x2r, where n and r are integers such that 16 ≤ n ≤ 31 and 
-3 ≤ r ≤ 4, i.e. a normalized binary floating point encoding with 1 sign bit, 4 bits of fraction and a 3-bit exponent. 
Note that this encoding does not include the value 0.0, however this value may be loaded using a floating-point 
move (register) instruction of the form FMOV Sd,WZR. 
FMOV Sd, #fpimm 
Single-precision floating-point move immediate Sd = fpimm.  
FMOV Dd, #fpimm 
Double-precision floating-point move immediate Dd = fpimm.

Reply:
Hmm... that's interesting.

Um, I appreciate your effort but I don't feel this is the right place to talk.
[riscv/riscv-isa-manual] is probably not the best place to discuss but at least it's better than here.